pkg/trie/cache/inmemory/lru_cache.go

```diff
@@ -0,0 +1,49 @@
+// Copyright 2024 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package inmemory
+
+import (
+	"github.com/karlseguin/ccache/v3"
+)
+
+// ccache uses 350 bytes of overhead per entry
+const cacheValueOverheadSize = 350
+
+// cacheValue is a helper alias over []byte to implement ccache.Sized
+type cacheValue []byte
+
+// Size returns the size of the cacheValue taking the overhead into account
+func (cv cacheValue) Size() int64 {
+	return int64(len(cv) + cacheValueOverheadSize)
+}
+
+// lruCache is an in-memory lru cache
+// consider that the values are deleted asyncronously so there is a chance that
+// the maxSize can be exceeded
+// we can use lru.GC() to force the deletion of the items that should be deleted
+type lruCache struct {
```

Comment:
differentiate what the difference is between this and the imported `LRUCache`?
```suggestion
type maxBytesLRUCache struct {
```

Reply:
Done!